kubechain/internal/controller/taskruntoolcall/taskruntoolcall_controller.go

```diff
@@ -678,7 +774,66 @@ func (r *TaskRunToolCallReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 			return ctrl.Result{}, err
 		}
 
+		approvalGranted := false
+		trtcNamespace := trtc.Namespace
+		var contactChannel *kubechainv1alpha1.ContactChannel
+		var apiKey string
+
 		if needsApproval {
+			contactChannel, err = r.getContactChannel(ctx, mcpServer, trtcNamespace)
+			if err != nil {
+				return r.setStatusError(ctx, kubechainv1alpha1.TaskRunToolCallStatusTypeErrorRequestingHumanApproval, "NoContactChannel", &trtc, err)
+			}
+
+			apiKey, err = r.getHumanLayerAPIKey(ctx, contactChannel.Spec.APIKeyFrom.SecretKeyRef.Name, contactChannel.Spec.APIKeyFrom.SecretKeyRef.Key, trtcNamespace)
+
+			if err != nil || apiKey == "" {
+				return r.setStatusError(ctx, kubechainv1alpha1.TaskRunToolCallStatusTypeErrorRequestingHumanApproval, "NoAPIKey", &trtc, err)
+			}
+		}
+
+		// Check if approved/rejected
+		if needsApproval && trtc.Status.Status == kubechainv1alpha1.TaskRunToolCallStatusTypeAwaitingHumanApproval {
+			client := r.HLClient.NewHumanLayerClient()
+			client.SetCallID(trtc.Status.HumanLayerCallId)
+			client.SetAPIKey(apiKey)
+			functionCall, _, _ := client.GetFunctionCallStatus(ctx)
+
+			status := functionCall.GetStatus()
+
+			// No response yet.
+			if status.RespondedAt.Get() == nil {
+				return ctrl.Result{RequeueAfter: 5 * time.Second}, nil
+			}
+
+			approvalGranted = status.GetApproved()
+
+			// approval rejected, end flow
+			if !approvalGranted {
+				trtc.Status.Status = kubechainv1alpha1.TaskRunToolCallStatusTypeError
+				trtc.Status.StatusDetail = "Tool execution rejected"
+				trtc.Status.Result = "Rejected"
+				trtc.Status.Phase = kubechainv1alpha1.TaskRunToolCallPhaseFailed
+
+				if err := r.Status().Update(ctx, &trtc); err != nil {
+					logger.Error(err, "Failed to update TaskRunToolCall status")
+					return ctrl.Result{}, err
+				}
+
+				return ctrl.Result{}, nil
+			}
+
+			if approvalGranted {
+				trtc.Status.Status = kubechainv1alpha1.TaskRunToolCallStatusTypePending
+				trtc.Status.StatusDetail = "Tool execution approved"
+				trtc.Status.Phase = kubechainv1alpha1.TaskRunToolCallPhaseSucceeded
+
+				// TODO: Do we want to update the status here and return? Right now we just let
+				// the rest of execution continue which allows tool execution to occur.
```

Comment:
now finally reading this and makes sense...you can slightly disregard the two comments near here